src/lake/Lake/Build/Actions.lean

```diff
@@ -39,7 +45,19 @@ def compileLeanModule (name : String) (leanFile : FilePath)
     ]
   }
 
-def compileO (name : String) (oFile srcFile : FilePath)
+def compileBcToO (name : String) (oFile srcFile : FilePath)
+  (target : String := "native") (clang : FilePath := "clang") : BuildM Unit := do
+  logStep s!"Compiling {name}"
+  createParentDirs oFile
+  proc {
+    cmd := clang.toString
+    args := #["-c",
+             "-march=" ++ target, -- '-march=<target>' must be a single string.
```

Comment:
We also need support to pull arguments into this function right?

Reply:
Yes, I held off against this for the first release, I would like to think this through a little bit more.